### PR TITLE
Added Animations

### LINKED ISSUE #2: Add animations to the clearing of cells via framer motion
I believe that it would be neat to have this game animate with framer motion. Go nuts!

### PR DESCRIPTION
**Pull request by Balaguru4580**
I've added some visual animations to improve playability.

closes #2

### CODE REVIEW

**Comment by BenMcH** [comment]
With these changes, I cannot see the minesweeper header. 
The page is also scrolling side to side now, which is undesirable. 
Can we disable the animations when the button is not active?

Do you think it'd be good to animate the bombs at all when they explode? Maybe a shake of the board? 

Overall, I like the spin in, but maybe we should start smaller and only scale to 1 to fix the header and scrolling issues. Not sure.

**Comment by Balaguru4580** [comment]
How is it compiling for you? I still keep getting those any type implicit errors.

Actually, I wasn't able to review any of the changes even with noimplicitAny set to true, which was not the case before I updated my fork.

Any help would be great.

**Comment by BenMcH** [comment]
What are the steps you're taking? I pulled, ran yarn install, and ran yarn start.

**Comment by Balaguru4580** [comment]
Oh. I just ran yarn by itself before, it's working now. Working on the changes now.

**Comment by Balaguru4580** [comment]
Alright, I've modified the animations to fit within the scale of 1 to avoid that horizontal scroller issue. I've added some other animations too.

**Comment by BenMcH** [comment]
I think we're getting really close @Balaguru4580 :smile: The only bug that I'm seeing now is that the losing animation only happens on the first loss. If you click reset and lose again, you should get the animation again. Once this is fixed, I'll gladly approve and merge. Great work! :tada:

**Comment by Balaguru4580** [comment]
Somehow that issue took me a while to figure out how to fix, but it is now fixed!

File: src/App.tsx [hunk 1]
```diff
@@ -160,6 +162,10 @@ function App() {
   }
 
   const size = getSize(board.numCols, board.numRows);
+  const boomshake = {
```
Comment (BenMcH) [inline src/App.tsx]:
```suggestion
  const boomShake = {
```

File: src/App.tsx [hunk 2]
```diff
@@ -160,6 +162,10 @@ function App() {
...
+    shake: {x: 10, transition: {type: 'spring', stiffness: 2000}},
```
Comment (BenMcH) [inline src/App.tsx]: I experimented with some keyframes and like these values to really make it chaotic for a couple of seconds

```suggestion
    shake: {x: [0, 20, -20, 10, -10, 0], y: [0, 20, -20, -10, 10, 0], transition: {type: 'spring', stiffness: 2000}},
```

**Review by BenMcH — COMMENTED** [review]
Other than the comments left, if we can remove the ts-ignores, I'd be happy to merge. I was just able to remove them and build your branch with the other proposed changes and didn't get compilation errors, so I'm not sure why they were needed originally.

File: src/App.tsx (hunk 2 above)
Comment (Balaguru4580) [reply src/App.tsx]: I just tried out the new values. Your solution works way better. It even allows for bypassing the 'slide back to normal' animation that I couldn't figure out how to remove.

Committing now.

**Comment by Balaguru4580** [comment]
As it turns out, it seems like I did an improper installation, which led to those compilation errors, which necessitated ts-ignore. The ts-ignore comments are removed now, after proper installation with yarn.

**Pull request merged**